open-metadata-implementation/adapters/open-connectors/rest-client-connectors/spring-rest-client-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/restclients/spring/SpringRESTClientConnector.java

```diff
@@ -41,49 +49,317 @@
+                        ksFile.close();
 
+                        keyManagers = kmFactory.getKeyManagers();
+                    }
+                }
+            }
+            else {
+                trustManagers = INSECURE_MANAGER;
+                keyManagers = null;
+            }
+
+            // Get a good seed for the crypt
+            SecureRandom random = new SecureRandom();
+            random.setSeed(System.currentTimeMillis());
+
+            // Initialize the ssl context now we know our key/cert managers
+            sc.init(keyManagers, trustManagers, random);
+
+            // Get a custom httpclient - scope to Egeria rest API calls & do not impact other usages
+            HttpClientBuilder httpBuilder = HttpClientBuilder.create();
+            httpBuilder.useSystemProperties();
+            httpBuilder.setSSLContext(sc);
+
+            // Host verification is set within the http builder - not relevant if SSL verification is being skipped
+            if (sslVerify)
+                httpBuilder.setSSLHostnameVerifier(new DefaultHostnameVerifier());
+            else
+                httpBuilder.setSSLHostnameVerifier(new NoopHostnameVerifier());
+
+            CloseableHttpClient httpClient;
+            httpClient = httpBuilder.build();
+
+            HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
+            requestFactory.setHttpClient(httpClient);
+
+            // TODO: Now use restTemplate - consider what caching can be done in case performance above is comprimized
+            restTemplate = new RestTemplate(requestFactory);
+
+            /*
+             * Rather than creating a RestTemplate directly, the RestTemplateBuilder is used so that the
+             * uriTemplateHandler can be specified. The URI encoding is set to VALUES_ONLY so that the
+             * '+' character, which is used in queryParameters conveying searchCriteria, which can be a
+             * regex, is encoded as '+' and not converted to a space character.
+             * Prior to this change a regex containing a '+' character would be split into two space
+             * separated words. For example, the regex "name_0+7" (which would match name_07, name_007,
+             * name_0007, etc) would be sent to the server as "name_0 7".
+             */
+            DefaultUriBuilderFactory builderFactory = new DefaultUriBuilderFactory();
+            builderFactory.setEncodingMode(DefaultUriBuilderFactory.EncodingMode.VALUES_ONLY);
+            restTemplate.setUriTemplateHandler(builderFactory);
+
+            /* Ensure that the REST template always uses UTF-8 */
+            List<HttpMessageConverter<?>> converters = restTemplate.getMessageConverters();
+            converters.removeIf(httpMessageConverter -> httpMessageConverter instanceof StringHttpMessageConverter);
+            converters.add(0, new StringHttpMessageConverter(StandardCharsets.UTF_8));
```

Comment:
Why to limit?

Reply:
This has been highlighted after some code moving around in the method - but it wasn't introduced by the PR. It came in   #1872